006-strimzi-ui.md

```diff
@@ -0,0 +1,129 @@
+- Integrating the UI with Strimzi itself
+- A Kafka backend server for Kafka data/requests the UI server can call
+- UI server configurability/capability - including aspects such as;
+  - Transport security
+  - User session, authentication and authorization capabilities
+  - Cluster metadata - listener addresses, number of brokers, etc
+  - Maintenance/tracing support
+- UI client side capabilities
+  - State design
+  - Configuration/capability discovery from the server
+  - Routing and navigation logic
+  - Maintenance/tracing capabilities
+
+I am more than happy to elaborate and collaborate on any of these points here, and would suggest these (and any follow items identified while discussing this proposal) are completed before any client side UI work begins.
+
+#### Topic list page
+
+Given the prerequisites are satisfied, the first user task I would suggest is implemented is a listing of all the topics (along with partition and replica information per topic) in a given cluster. This will exercise all the prerequisites, as well as offer a user of Strimzi a new way to interact with their deployment. Using a [gherkin](https://cucumber.io/docs/gherkin/reference/) behavioural syntax to describe the capability, it would:
+
+```
+Scenario: Seeing all available topics:
+
+Given I am a user of the UI
+When I navigate to the topic listing
+Then see all topics in my Kafka cluster
+
+Scenario: Viewing a specific topic (when it exists in the cluster):
+Given I am a user of the UI
+When I navigate to the topic listing
+And I filter for topic 'SampleTopic'
+Then I see topic 'SampleTopic' in the topic list
+And Topic 'SampleTopic' has '3' partitions and '2' replicas
+
+Scenario: Viewing a specific topic (when it does not exist in the cluster):
+Given I am a user of the UI
+When I navigate to the topic listing
+And I filter for topic 'SampleTopic'
+Then I am told topic 'SampleTopic' does not exist in the cluster
+```
+
+#### Follow on capabilities
+
+Additional Kafka capabilities which could be added later include (but would not be limited to):
+
+- Create, update and delete topics on the topics list page (recontextualised as a Topics page - as per mock up)
+- View Consumer group status for a given topic, or the whole cluster
+- View the Brokers in a given cluster, their configuration, and where appropriate, allow modification of broker configuration
+- Provide details of Kafka listener/bootstrap addresses, along with sample configuration, to allow streamline client connectivity
+
+If desired, additional Strimzi provided capabilities, such as Kafka Users, could be leveraged/added/managed through this UI also (for example, allowing access to the UI and its capabilities, general CRUD updates to them etc). 
+
+Finally, one capability which may be of interest (which may have bearing on how this is surfaced in Strimzi itself) is having a single UI, which can manage multiple Kafka clusters deployed via the Strimzi operator.
```

Comment:
with this comment are you assuming that if the Strimzi operator deploys N Kafka cluster in N different namespaces, each Kafka cluster gets its own UI server + backend server? It's quite related to the comment I made above.

Reply:
I am just about to push a commit with a proposed deployment approach which demonstrates how this could be implemented. My thought is there is one UI (and UI server), configured to be aware of N Kafka clusters in N namespaces.

Reply:
And what about the backend? Is it one for namespace where a Kafka cluster lives?